framework/Models/HybridModel.py

```diff
@@ -0,0 +1,621 @@
+    """
+      This method describes the types of input accepted with a certain role by the model class specialization
+      @ In, None
+      @ Out, None
+    """
+    pass
+
+  def __init__(self,runInfoDict):
+    """
+      Constructor
+      @ In, runInfoDict, dict, the dictionary containing the runInfo (read in the XML input file)
+      @ Out, None
+    """
+    Dummy.__init__(self,runInfoDict)
+    self.modelInstance            = None
+    self.cvInstance               = None
+    self.targetEvaluationInstance = None
+    self.tempTargetEvaluation     = None
+    self.romsDictionary        = {}      # dictionary of models that is going to be employed, i.e. {'romName':Instance}
+    self.romTrainStepSize      = 1        # the step size for rom train
+    self.romTrainStartSize     = 10       # the initial size of training set
+    self.romTrainMaxSize       = 1.0e6     # the maximum size of training set
+    self.romValidateSize       = 10       # the size of rom validation set
+    self.counter               = 0        # record the number of model runs
+    self.romTrained            = False    # True if all roms are trained
+    self.sleepTime             = 0.005    # waiting time before checking if a run is finished.
+    self.romConverged          = False    # True if all roms are converged
+    self.romValid              = False    # True if all roms are valid for given input data
+    self.romConvergence        = 0.01
+    self.tempOutputs           = {}
+    self.optimizerTypes        = ['SPSA'] # list of types of optimizer
+    self.modelSelection        = 'CrowdingDistance'
+    self.existTrainSize        = 0
+    self.printTag              = 'HYBRIDMODEL MODEL' # print tag
+    self.createWorkingDir      = False
+    # assembler objects to be requested
+    self.addAssemblerObject('Model','1',True)
+    self.addAssemblerObject('ROM','n')
+    self.addAssemblerObject('CV','1')
+    self.addAssemblerObject('TargetEvaluation','1')
+
+  def localInputAndChecks(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None
+    """
+    Dummy.localInputAndChecks(self, xmlNode)
+    for child in xmlNode:
+      if child.tag == 'settings':
+        self.__readSettings(child)
+      elif child.tag == 'Model':
+        self.modelInstance = child.text.strip()
+        if child.attrib['type'] == 'Code':
+          self.createWorkingDir = True
+      elif child.tag == 'CV':
+        self.cvInstance = child.text.strip()
+      elif child.tag == 'TargetEvaluation':
+        self.targetEvaluationInstance = child.text.strip()
+      elif child.tag == 'ROM':
+        romName = child.text.strip()
+        # 'useRom': [converged, validated]
+        self.romsDictionary[romName] = {'Instance':None,'Converged':False,'Valid':False}
```

Comment:
I thought for the models we were using the inputParameters system.
@joshua-cogliati-inl ?

Reply:
fixed.